lib/ferrum/errors.rb

```diff
@@ -27,11 +27,17 @@ def initialize(url, pendings = [])
   end
 
   class TimeoutError < Error
-    def message
-      "Timed out waiting for response. It's possible that this happened " \
-        "because something took a very long time (for example a page load " \
-        "was slow). If so, setting the :timeout option to a higher value might " \
-        "help."
+    attr_reader :pending_urls
+
+    def initialize(pending_urls = [])
+      @pending_urls = pending_urls
+
+      message = "Timed out waiting for response. It's possible that this happened " \
+                "because something took a very long time (for example a page load " \
+                "was slow). If so, setting the :timeout option to a higher value might " \
+                "help."
+      message = "#{message}\n Connections still pending:\n #{pending_urls.join(', ')}" unless pending_urls.empty?
```

Comment:
This isn't a critique of your code here (because I use unless a lot) - but I read this the other day:
_Read This Post 'Unless' You're Not A Ruby Developer_

https://news.ycombinator.com/item?id=33965933

Reply:
Do you want to override the whole message string, or just append to it if there are pending urls? (it's replacing it atm)

Reply:
Maybe something like:
```suggestion
  if !pending_urls.empty?
      message =+ "#{message}\n Connections still pending:\n #{pending_urls.join(', ')}" unless pending_urls.empty?
    end
```